bootstrap/main.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+
+import json
+import os
+import pathlib
+import shutil
+import sys
+import time
+from typing import Any
+from warnings import warn
+
+import docker
+
+client = docker.from_env()
+low_level_api = docker.APIClient(base_url="unix://var/run/docker.sock")
+
+DEFAULT_FILE_PATH = pathlib.Path("/bootstrap/startup.json.default").as_posix()
+DOCKER_CONFIG_PATH = pathlib.Path("/root/.config/companion/startup.json").as_posix()
+
+HOST_CONFIG_PATH = os.environ.get("COMPANION_CONFIG_PATH", None)
+
+CORE_CONTAINER_NAME = "companion_core"
+
+
+def ensure_dir(file_path: str) -> None:
+    """Makes sure that the file path exists
+
+    Args:
+        file_path (str): path to check/make
+    """
+    directory = pathlib.Path(file_path).parent
+    os.makedirs(directory, exist_ok=True)
+
+
+def stop_old_container() -> None:
+    """ Looks for an old core container and stops it if found"""
+    try:
+        old_container = client.containers.get(CORE_CONTAINER_NAME)
+        old_container.stop()
+        old_container.remove()
+    except docker.errors.NotFound:
+        pass
+
+
+def overwrite_config_file_with_defaults() -> None:
+    """Overwrites the config file with the default configuration"""
+    shutil.copy(DEFAULT_FILE_PATH, DOCKER_CONFIG_PATH)
+
+
+def read_config_file() -> Any:
+    """Tries to read the config file
+
+    Returns:
+        Any: Json data of startup.json
+    """
+
+    # Tries to open the current file
+    config = {}
+    try:
+        with open(DOCKER_CONFIG_PATH) as config_file:
+            config = json.load(config_file)
+    except Exception:
+        warn("unable to read startup.json file, reverting to defaults...")
```

Comment:
can we f-string the warning with the exception type ?

Reply:
sure, done